### PR TITLE
node-1119-skill-map-search-api

### PR DESCRIPTION
**Pull request by chadiegilnerubia**
Ticket ID:
Create and Integrate API for Skill Map Search screen
[#1119](https://github.com/nerubia/kh360-react/issues/1119)

### CODE REVIEW

File: src/services/skill-map-search-service.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+/* eslint-disable no-console */
```
Comment (jatnerubia) [inline src/services/skill-map-search-service.ts:1]: remove this ^^

File: src/services/skill-map-search-service.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import * as SkillRepository from "../repositories/skill-repository"
+
+export const getAllByFilters = async (name: string, skill: string, page: string) => {
+  const itemsPerPage = 10
+  const parsedPage = parseInt(page)
+  const currentPage = isNaN(parsedPage) || parsedPage < 0 ? 1 : parsedPage
+
+  const allRecentRating = await SkillMapSearchRepository.getLatestSkillMapRating()
+
+  // Filter by name
```
Comment (jatnerubia) [inline src/services/skill-map-search-service.ts:12]: remove this ^^

File: src/services/skill-map-search-service.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  const filteredResultsByName = allRecentRating.filter((result) => {
+    if (name !== undefined) {
+      const user = result.users
+      return user?.first_name?.toLowerCase().includes(name.toLowerCase()) ?? false
+    }
+    return true
+  })
+
+  // Filter by skill
+  const filteredResultsBySkill = []
```
Comment (jatnerubia) [inline src/services/skill-map-search-service.ts:22]: remove this ^^

File: src/services/skill-map-search-service.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      )
+      if (matchesSkill.some(Boolean)) {
+        filteredResultsBySkill.push(result)
+      }
+    } else {
+      filteredResultsBySkill.push(result)
+    }
+  }
+
+  // Implement pagination
```
Comment (jatnerubia) [inline src/services/skill-map-search-service.ts:41]: remove this ^^

**Review by jatnerubia — CHANGES_REQUESTED** [review]
minor changes ^^

File: src/services/skill-map-search-service.ts (hunk 1 above)
Comment (chadiegilnerubia) [reply src/services/skill-map-search-service.ts]: done :3

**Review by jatnerubia — APPROVED** [review]
Cool

**Pull request merged**